### ISSUE TITLE
Why does a fresh dokuwiki tar-ball contain files to delete?

### ISSUE THREAD

**Issue by zabarabab** [splitbrain/dokuwiki#1348]
Download and extract tar-ball:

```
wget -O - http://download.dokuwiki.org/src/dokuwiki/dokuwiki-stable.tgz | tar xz
```

Check for files to delete:

```
cd dokuwiki-2015-08-10a
egrep -v '^($|#)' data/deleted.files | while read file; do test -e "$file" && echo "$file"; done
```

Turns out, that these files have been deleted in older versions, an now they are used again. Maybe data/deleted.files should be split into one file per release?

**Comment by Klap-in** [comment]
Which files are still included?

**Comment by zabarabab** [comment]
lib/plugins/authad/lang/lv/settings.php
    lib/plugins/authldap/lang/lv/settings.php
    lib/plugins/authmysql/lang/fi/settings.php
    lib/plugins/authmysql/lang/lv/settings.php
    lib/plugins/authpgsql/lang/it/settings.php
    lib/plugins/authpgsql/lang/lv/settings.php
    lib/plugins/authpgsql/lang/pl/settings.php
    lib/plugins/popularity/lang/hr/lang.php
    lib/plugins/revert/lang/hr/lang.php
    lib/plugins/usermanager/lang/hr/lang.php

**Comment by bug** [comment]
Not surprising this happens for translated strings. Maybe time to switch to "we provide these files" (manifest like, git ls-files) rather than a "we no longer provide these files". Opinions?

**Comment by splitbrain** [comment]
A manifest wouldn't help with deleting unused files. We cannot simply delete everything we don't know 😉

I think we need to adjust our procedure creating the list of deleted files to accommodate the case where files are readded later on.

**Comment by selfthinker** [comment]
Why do translations need to be part of `deleted.files` anyway? Unless it's from a deleted plugin, there is no need...

**Comment by splitbrain** [comment]
AFAIR those files were deleted because the code changed and these translation files were dropped. It seems they were readded later. But maybe I'm completely wrong. Haven't looked too deeply into that, yet.

**Comment by splitbrain** [comment]
The problem is how we generate the list of deleted files. We compare the list of files from the previous release with the list of files from the new release. All files that no longer exist in the new release will be added to the deleted.files list.

The problem is with language files we delete because they contain bogus translations. Eg. they are just straight copies from English. It's true that these files are gone, but they will come back again as soon someone creates a new translation.

For now I cleaned up the list of files manually. I'm not sure how to best solve the underlying problem.

**Comment by splitbrain** [comment]
should have been fixed with 8e72a512e7101b37d40bef012f1e717131622a91
